crates/bridge-proof/protocol2/src/tx_inclusion_proof.rs

```diff
@@ -0,0 +1,186 @@
+use bitcoin::{block::Header, hashes::Hash, Transaction};
+use borsh::{BorshDeserialize, BorshSerialize};
+use strata_bridge_primitives::bitcoin::BitcoinTx;
+use strata_primitives::{
+    buf::Buf32,
+    hash::sha256d,
+    l1::{TxIdComputable, TxIdMarker, WtxIdMarker},
+};
+use strata_proofimpl_btc_blockspace::block::witness_commitment_from_coinbase;
+use strata_state::l1::{L1TxInclusionProof, L1TxProof, L1WtxProof};
+
+/// A transaction along with its [L1TxInclusionProof], parameterized by a `Marker` type
+/// (either [`TxIdMarker`] or [`WtxIdMarker`]).
+///
+/// This struct pairs the actual Bitcoin [`Transaction`] with its corresponding proof that
+/// its `txid` or `wtxid` is included in a given Merkle root. The proof data is carried
+/// by the [`L1TxInclusionProof`].
+#[derive(Debug, Clone, BorshSerialize, BorshDeserialize)]
+pub(crate) struct L1TxWithIdProof<T> {
+    /// The transaction in question.
+    tx: BitcoinTx,
+    /// The Merkle inclusion proof associated with the transaction’s [`Txid`](bitcoin::Txid) or
+    /// [`Wtxid`](bitcoin::Wtxid).
+    proof: L1TxInclusionProof<T>,
+}
+
+impl<T: TxIdComputable> L1TxWithIdProof<T> {
+    // Ignored for now. This is meant to be called from elsewhere to generate to the format to be
+    // used by the prover
+    #[allow(dead_code)]
```

Comment:
```suggestion
    #[expect(dead_code)]
```

`expect` will produce an error if this is not in fact `dead_code`. This will help clean things up more easily when this gets used.

Reply:
I was not aware of this! TIL.

Reply:
This is pretty new -- stabilized in [1.81.0](https://github.com/alpenlabs/strata-bridge/blob/ea2fa37b0982ba9753ac902b55e5b15e89d8faed/Makefile#L46-L54).